### ISSUE TITLE
When will the code be released?

### ISSUE THREAD

**Issue by mikigom** [khalooei/ALOCC-CVPR2018#1]
(What the title says)

**Comment by khalooei** [comment]
Dear Sir/madam
We are busy now and try to publish it  as soon as possible.
Best regards

**Comment by khalooei** [comment]
Dear Sir/madam
We are busy now and try to publish clean code as soon as possible.
Please feel free to contact me if you have any questions.
Best regards

**Comment by zjzzzz** [comment]
Dear Sir:
        I am very interested in your paper.Thus I hope the code can be released soon,so that I may can have a try.☺☺☺
Best wishes

**Comment by legolas123** [comment]
I do not think with the details given in paper, it works. I have implemented this with many configurations and trained them with different parameter settings. I know that implementation is not wrong because autoencoded images are correctly being generated. I think there are few crucial things not mentioned in paper. We will have to wait for author's code release in order to verify whether the idea mentioned in paper really works or not.

**Comment by robertmash2** [comment]
I too have implemented the algorithm as described in the paper, but with some guesses in the loss optimization specifics.  So far, autoencoding works, but R(D(x)) is worse than D(x).

**Comment by legolas123** [comment]
Yeah, so you too facing the same problem. It increases the likelihood that many crucial things are not mentioned in the paper. I hope the CVPR paper is not incorrect in reporting the results.

**Comment by kings123456** [comment]
I have implemented this paper. In many of cases i've achieved better result in comparison of the paper. The idea works but it is very very depend on when you stop training of GAN, And howmuch noise are added to input of R network. A related but more interesting paper about this topic is :https://arxiv.org/pdf/1807.02588.pdf which solve this problem using adversarial learning

I could share my implementation, if you need please send an email for me.  I am newbie on GIT and may be i miss your request.  fafarde86@gmail.com

**Comment by Sabokrou** [comment]
@legolas123 @robertmash2:
 What is your stop criteria for training of of generative adversarial nets? If you stop in a wrong time, D(R(X)) will be worse than D(X). As mentioned in paper, "Stopping the training too early leads to immature learned network weights, while overtraining the networks confuses the R module and yields undesirable outputs."

**Comment by Sabokrou** [comment]
@kings123456 : What is your implementation framework?Tensor-flow or Torch?  Could you please send your code for me?

**Comment by legolas123** [comment]
@Sabokrou I stopped it by applying condition on the euclidean distance between generated and real images as per mentioned in paper. But I think problem is not that. The GAN part of the network does not seem to converge because there could be overfitting due to the large network(mentioned in paper) and using very less number of images(150*3 classes). 
Can you please share your code and details for the implementation on caltech-256 dataset?

**Comment by Sabokrou** [comment]
Is your outlier samples distorted? Did you check?  At this time, we couldn't release the code. 
Simply you can use validation samples to track behavior of two networks in training phase.

**Comment by legolas123** [comment]
Yep, exactly I checked that. And they are being reconstructed as smoothly as inliers. That is when I was sure something is wrong. I have taken 150 images from three different classes. And trained the network with three different sizes(32, 64, 128). I am resizing the whole image to the input size of network. Am I doing something wrong here. And did you also use only 150*3 number of images for the experiment with three classes as inliers?

**Comment by legolas123** [comment]
@Sabokrou Can you atleast outline the implementation details for caltech-256. Input size of the architecture, number of images, architecture of network, any special type of GAN you used etc.

**Comment by Sabokrou** [comment]
Yes,  i will share details with you. 


Sent from my Samsung Galaxy smartphone.

**Comment by wuchihsu** [comment]
Is it possible that discriminator can recognize samples from reconstructor very easily so even the outliers have been distorted their D(R(X)) cannot be separated from inliers' ?

**Comment by khalooei** [comment]
To Whom it May Concern,

I push the preliminary version of the code on grayscale databases.

Thanks for your consideration in advance.
Best regards,
Mohammad
